packages/api/src/external/carequality/patient.ts

```diff
@@ -56,23 +94,72 @@ export async function discover(
   }
 }
 
-export async function handleDiscoverResponse(
-  patientDiscoveryResponse: PatientDiscoveryResponse
+export async function handlePatientDiscoveryResults(
+  patient: Patient,
+  pdResults: PatientDiscoveryResult[]
 ): Promise<void> {
-  const { cxId, patientId, xcpdPatientId, patientMatch, gateway } = patientDiscoveryResponse;
-
-  if (patientMatch) {
-    await setCarequalityId({
-      patientId,
-      cxId,
-      carequalityPatientId: xcpdPatientId?.id ?? "",
-      carequalityPatientSystemId: xcpdPatientId?.system ?? "",
-    });
-  } else {
-    // WE SHOULD EXPECT THIS TO HAPPEN SO SHOULD WE ADD SENTRY MESSAGE?
-    // DONT WANT TO BOMBARD ALERTS CHANNEL
-    console.log(
-      `No patient match in IHE Gatway for patientId ${patientId}, cxId ${cxId}, gatewayUrl ${gateway.url}, gatewayId ${gateway.oid}`
-    );
+  if (pdResults.length === 0) {
+    console.log(`No patient discovery results found for patientId: ${patient.id}`);
+    return;
   }
+  const { id, cxId } = patient;
+
+  const cqLinks = buildCQLinks(pdResults);
+  const pdCreate: PatientCQDataCreate = {
+    id,
+    cxId,
+    data: { links: cqLinks },
+  };
+
+  if (cqLinks.length) await createOrUpdatePatientCQData(pdCreate);
+}
+
+export function buildCQLinks(pdResults: PatientDiscoveryResult[]): CQLink[] {
+  return pdResults.flatMap(pd => {
+    const id = pd.data.xcpdPatientId?.id;
+    const system = pd.data.xcpdPatientId?.system;
+    if (!id || !system) return [];
+    return {
+      patientId: id,
+      systemId: system,
+      ...pd.data.gateway,
+    };
+  });
+}
+
+function startTimeout() {
+  const timeout = PATIENT_DISCOVERY_TIMEOUT.asMilliseconds();
+  return new Promise<string>(resolve => {
+    setTimeout(() => {
+      const msg = `Patient discovery reached timeout after ${timeout} ms`;
+      resolve(msg);
+    }, timeout);
+  });
+}
+
+function checkNumberOfResults(
+  raceControl: RaceControl,
+  requestId: string,
+  numberOfGateways: number
+): Promise<string> {
+  return new Promise(resolve => {
+    const checkAndResolve = () => {
+      isPDComplete(requestId, numberOfGateways).then(isComplete => {
```

Comment:
NABD if really needed, but we prefer/use await/async across the code and avoid `.then()`